core/tests/puppeteer-acceptance-tests/utilities/common/puppeteer-utils.ts

```diff
@@ -724,19 +724,29 @@ export class BaseUser {
               path.basename(dirName)
             )
           : path.join(testPath, dirName, 'diff-snapshots'),
+        storeReceivedOnFailure: true,
+        customReceivedPostfix: '',
+        customReceivedDir: __dirname.startsWith('/home/runner')
+          ? path.join(
+              '/home/runner/work/oppia/oppia/core/tests/puppeteer-acceptance-tests/diff-snapshots',
+              path.basename(dirName)
+            )
+          : path.join(testPath, dirName, 'diff-snapshots'),
       });
       if (typeof newPage !== 'undefined') {
         await newPage.close();
       }
     } catch (error) {
+      var ErrorMessage = error.message;
       if (__dirname.startsWith('/home/runner')) {
```

Comment:
I think, better to have a variable that stores whether the run is happening on CI or locally. Then use that variable to make decisions about what to do.